File: aula5/Em_java/Guerreiro.java
public class Guerreiro extends Personagem{
    private int defesa;
    public Guerreiro(String nome, int vida, int nivel, int forca){
        super(nome, vida, nivel, forca);
        this.setDefesa(5);
    }
    public int getDefesa() {
        return defesa;
    }
    public void setDefesa(int defesa){
        if(defesa < 0){
            throw new IllegalArgumentException("Defesa nao pode ser negativa: " + defesa);
        }
        this.defesa = defesa;
    }
    @Override
    public String habilidade(){
        return "Soco poderoso";
    }
    @Override
    public void atacar(Personagem alvo) throws SemMana{
        super.atacar(alvo);
    }
    @Override
    public void ficha(){
        System.out.println("\nGuerreiro");
        super.ficha();
        System.out.println("Defesa: " + this.defesa);
    }
}

File: aula5/Em_python/Guerreiro.py
from Personagem import Personagem
class Guerreiro(Personagem):
    def __init__(self, nome: str, vida: int, nivel: int, forca: int):
        super().__init__(nome, vida, nivel, forca)
        self._defesa = 5
    def habilidade(self) -> str:
        return "corte_duplo"
    @property
    def defesa(self):
        return self._defesa
    @defesa.setter
    def defesa(self, valor: int):
        if valor < 0:
            print("A defesa não pode ser negativa.")
        else:
            self._defesa = valor
    def ficha(self):
        print("Ficha do Guerreiro:")
        super().ficha()
        print(f"Defesa: {self._defesa}")